docker-compose.yml

```diff
@@ -0,0 +1,33 @@
+version: '3'
+services:
+  db:
+    image: postgres:9.6.22
+    environment:
+      POSTGRES_DB: ${DATABASE_NAME}
+      POSTGRES_USER: ${DATABASE_USER}
+      POSTGRES_PASSWORD: ${DATABASE_PASS}
+    healthcheck:
+      test: ["CMD-SHELL", "pg_isready -U ${DATABASE_USER} -d ${DATABASE_NAME}"]
+      interval: 5s
+      timeout: 5s
+      retries: 5
+    ports:
+      - "${DATABASE_PORT}:${DATABASE_PORT}"
+    volumes:
+      - db-data:/var/lib/postgresql/data/
+
+  livedata:
+    image: live_data:dev
```

Comment:
This I don't understand. Can you clarify the meaning of this line for me? I though you'd be building the image with the Dockerfile instead of what it looks like importing an already created image.

Reply:
This docker-file is intended as a way to demonstrate how to integrate the live_data_server into an existing system (i.e. it will need some env var defined as well as a predefined db that has the necessary table entries).
Most dev's would build the image locally, and the tag `:dev` is a common tag used to describe the local developed image.
Adding the building info here doesn't really help as we will not use docker-compose to build the image.

Reply:
If you don't plan on building the image with `docker-compose` then you'll need to [pass the secret database account at `docker build` time](https://vsupalov.com/docker-build-pass-environment-variables/). I suggest to store the `docker-build` command as a Makefile target because it will probably be a lengthy command.

Reply:
You don't need to pass the db info during the building time.
A simple **docker build -t TAGNAME .** is sufficient.

Reply:
It is as you say provided you leave the database admin account secrets in the Dockerfile. If you remove then, then you need to pass then as arguments to `docker build`.

Reply:
I am a little bit lost at the moment, so here is what I recommend:
- leave everything as it is (this is just for documentation purpose according to the original enabler).
  - the actual artifact needed is the image, which is already in Gitlab
- build a local image of the live_data_server with: `docker build -t live_data:dev .`
- spin up the virtual network with docker-compose so that we can verify that the live_data server can talk to the external database via `docker-compose up`
- if needed, go to `localhost:9999/admin`, and you should get 400 error as the livedata server does not have a valid admin page enabled
  - this is the current state of this app, and we are not refactoring/modernizing this web app according to the original Gitlab enabler.

Reply:
I'm happy with everything except that initialization of variables DATABASE_USER, DATABASE_PASS, DATABASE_PORT, and DATABASE_HOST should be removed from the `.env` and `DockerFile` files.
@peterfpeterson , sorry but it's up to you to break the tie :smile: